### DISCUSSION TITLE
Cart query returns `null` only on Heroku

### DISCUSSION THREAD

**Discussion by duailibe** [Shopify/storefront-api-feedback#152] [Help] [answered]
Hi! We have been running our headless site hosted on Heroku for a few weeks now and today, all of a sudden, every query of `{ cart(id: $cartId) { ... } }` was returning `null`. Strangely, this did not happen when I was running the app locally.

I was able to reproduce the problem by querying the API directly with `curl` (see screenshots below):

| Local | Heroku |
|---|---|
| ![local](https://user-images.githubusercontent.com/1574588/200954874-05239a41-3afc-4f7f-b27a-4748c2fb77a6.png) | ![heroku](https://user-images.githubusercontent.com/1574588/200955147-ee7001cf-bebc-4ad8-b664-8ef377aef977.png) |

It's hard too see like that but those are the exact same query for the same ID, works locally, returns `null` on Heroku.

What's more strange, was that if I added or removed a single field or even change the fields order from this query, the query responds correctly.

A request ID of a failing query: 2042d68d-956b-40fb-9868-b7086c34b070

**Comment by johncraigcole** [comment, marked as answer]
For the benefit of the community, I just wanted to provide a little context and confirm this was resolved. In 2022 we've been moving Storefront API queries to the edge to improve performance. We uncovered an intermittent issue where requests were routing to a region where the cart did not exist and returned a null cart response. The issue was identified, and a fix has been pushed to production.

**Reply by arobbins** [reply]
I think I'm getting the same error right now.

What's strange, the mutation works with a different selling plan ID on the same variant. So it seems to be isolated to this specific selling plan.

![screenshot-working](https://github.com/Shopify/storefront-api-feedback/assets/961947/57a85f32-4ad0-4459-986f-9ad05ab390d5)
![screenshot-not-working](https://github.com/Shopify/storefront-api-feedback/assets/961947/ff3c12ef-0baa-4586-80fe-664b5d0ee87c)

**Reply by arobbins** [reply]
Any idea if my above issue is related to this or not?

I just ran into this issue again today.
